torchtext/experimental/datasets/text_classification.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import torch
+import io
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import ngrams_iterator
+from torchtext.data.utils import get_tokenizer
+from torchtext.vocab import build_vocab_from_iterator
+from torchtext.vocab import Vocab
+from torchtext.datasets import TextClassificationDataset
+
+URLS = {
+    'IMDB':
+        'http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz'
+}
+
+
+def _create_data_from_iterator(vocab, iterator, removed_tokens):
+    for cls, tokens in iterator:
+        yield cls, iter(map(lambda x: vocab[x],
+                        filter(lambda x: x not in removed_tokens, tokens)))
+
+
+def _imdb_iterator(key, extracted_files, tokenizer, ngrams, yield_cls=False):
+    for fname in extracted_files:
+        if 'urls' in fname:
+            continue
+        elif key in fname and ('pos' in fname or 'neg' in fname):
+            with io.open(fname, encoding="utf8") as f:
+                label = 1 if 'pos' in fname else 0
+                if yield_cls:
+                    yield label, ngrams_iterator(tokenizer(f.read()), ngrams)
+                else:
+                    yield ngrams_iterator(tokenizer(f.read()), ngrams)
+
+
+def _generate_data_iterators(dataset_name, root, ngrams, tokenizer, data_select):
```

Comment:
I don't think you need to repeat all of this, but you can use it from the torchtext/datasets/text_classification file